embedded-graphics/src/primitives/line/mod.rs

```diff
@@ -86,49 +149,216 @@ impl Line {
+        let mut it = ParallelsIterator::new(self, thickness.saturating_cast(), alignment);
         let reduce =
             it.parallel_parameters.position_step.major + it.parallel_parameters.position_step.minor;
 
         let mut left = (self.start, ParallelLineType::Normal);
         let mut right = (self.start, ParallelLineType::Normal);
 
-        loop {
-            if let Some((bresenham, reduce)) = it.next() {
-                right = (bresenham.point, reduce);
-            } else {
-                break;
+        match alignment {
+            StrokeAlignment::Center => loop {
+                if let Some((bresenham, reduce)) = it.next() {
+                    right = (bresenham.point, reduce);
+                } else {
+                    break;
+                }
+
+                if let Some((bresenham, reduce)) = it.next() {
+                    left = (bresenham.point, reduce);
+                } else {
+                    break;
+                }
+            },
+            // Left side
+            StrokeAlignment::Outside => {
+                if let Some((bresenham, reduce)) = it.last() {
+                    left = (bresenham.point, reduce);
+                }
             }
-
-            if let Some((bresenham, reduce)) = it.next() {
-                left = (bresenham.point, reduce);
-            } else {
-                break;
+            // Right side
+            StrokeAlignment::Inside => {
+                if let Some((bresenham, reduce)) = it.last() {
+                    right = (bresenham.point, reduce);
+                }
             }
-        }
+        };
+
+        let left_start = left.0;
+        let right_start = right.0;
 
         let delta = self.end - self.start;
 
         let left_line = Line::new(
-            left.0,
-            left.0 + delta
+            left_start,
+            left_start + delta
                 - match left.1 {
                     ParallelLineType::Normal => Point::zero(),
                     ParallelLineType::Extra => reduce,
                 },
         );
 
         let right_line = Line::new(
-            right.0,
-            right.0 + delta
+            right_start,
+            right_start + delta
                 - match right.1 {
                     ParallelLineType::Normal => Point::zero(),
                     ParallelLineType::Extra => reduce,
                 },
         );
         (left_line, right_line)
     }
+
+    /// Sort line so point with smallest X coordinate is to the left.
+    fn sorted_x(&self) -> Self {
+        let (start, end) = if self.start.x > self.end.x {
+            (self.end, self.start)
+        } else {
+            (self.start, self.end)
+        };
+
+        Self::new(start, end)
+    }
+
+    /// Get the midpoint of the line.
+    pub fn midpoint(&self) -> Point {
+        self.start + (self.end - self.start) / 2
+    }
+
+    const fn coefficients(&self, other: &Self) -> (i32, i32, i32, i32, i32, i32, i32) {
```

Comment:
That would be great. I don't understand the `LinearEquation` code so probably can't do it myself, but it would be a lot cleaner to merge them if possible.